WalletWasabi/Crypto/ZeroKnowledge/KnowledgeOfDlog.cs

```diff
@@ -8,9 +8,9 @@
 
 namespace WalletWasabi.Crypto.ZeroKnowledge
 {
-	public class KnowledgeOfDiscreteLog : KnowledgeOfRepresentation
```

Comment:
i think this can just be removed instead of being renamed,  there's even a `param` based syntax sugar for the repr CreateProof, so you don't even need to make a list of arguments, this class doesn't add anything

Reply:
I like it there, makes it explicit that it's a special case of `KnowledgeOfRepresentation`, but I won't think twice removing this if it becomes an issue.